File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    priority_queue<int, vector<int>, greater<int>> q;
    int knums;

    KthLargest(int k, vector<int>& nums) {
        knums = k;
        for (int i : nums) {
            q.push(i);
        }

        while (q.size() > k) {
            q.pop();
        }
    }
    
    int add(int val) {
        q.push(val);

        while (q.size() > knums) {
            q.pop();
        }

        return q.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.py
class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        self.k = k
        self.numbers = nums
        heapq.heapify(self.numbers)
        while len(self.numbers) > k:
            heapq.heappop(self.numbers)
    
    def add(self, val: int) -> int:
        heapq.heappush(self.numbers, val)
        if len(self.numbers) > self.k:
            heapq.heappop(self.numbers)
        return self.numbers[0]
        


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)